[DEMO] 419 people picker react-window implementation

SemraSh opened the pull request:
This is an attempt to solve [this](https://github.com/elifesciences/elife-xpub/issues/419#issuecomment-409583197) issue using `react-window`.

hem-brahmbhatt committed:
WIP

hem-brahmbhatt committed:
added react-window demo

Commit:
Merge github.com:elifesciences/elife-xpub into 419-people-picker-windowing

Commit:
add react-window

hem-brahmbhatt committed:
fixed yarn.lock

hem-brahmbhatt committed:
Merge branch 'develop' into 419-people-picker-windowing

damnhipster commented:
**Performance flamegraph**
![screen shot 2018-12-11 at 15 03 38](https://user-images.githubusercontent.com/2573277/49809155-179d4e00-fd56-11e8-9452-0bde104cf747.png)

Notice how we're rendering a lot less pods even though the user can scroll through all 300-400 editors.

hem-brahmbhatt committed:
Merge branch 'develop' into 419-people-picker-windowing

hem-brahmbhatt committed:
updated yarn.lock

SemraSh committed:
Merge branch 'develop' into 419-people-picker-windowing

SemraSh commented:
### Findings of implementing `react-window`

With this implementation the required functionality is working perfectly fine as @damnhipster described above, but we have difficulties with adding styles. 

#### Behaviour on the browser
![react-window-attempt](https://user-images.githubusercontent.com/32770804/49950902-a219b500-fef0-11e8-808c-7a4b7e80c206.gif)

##### Problems
1. The scroll is not effected outside the PeoplePickerGrid component which holds the react-window's `Grid`. 
The only reason I can scroll on the right hand side is because of this line `width={window.innerWidth}` I've added to avoid the scrolling window to be in the middle of the page. I've added that thinking the expansion will be to both sides, which was solving the problem of being able to scroll just inside that component. 

I have another attempt to implement the older version of react-window called rect-virtualized [here](https://github.com/elifesciences/elife-xpub/pull/1195), as that version has WindowScroller component which is allowing scrolling anywhere in the page and has the same behaviour as react-window.

hdrury1 commented:
Now investigating alternatives to `react-window`, in particular paginating the API. Given what we don't know about `react-window` it is a risky dependency, and the work needed to override the styling is proving an obstacle.
